### PR TITLE
[SPARK-10969] [Streaming] [Kinesis] Allow specifying separate credentials for Kinesis and DynamoDB

### PR DESCRIPTION
**Pull request by kaklakariada**
`KinesisUtils.createStream()` allows specifying only one set of AWS credentials that will be used by Amazon Kinesis Client Library (KCL) for accessing Kinesis, DynamoDB and CloudWatch.
#### Motivation

In a scenario where one needs to read from a Kinesis Stream owned by a different AWS account the user usually has minimal rights (i.e. only read from the stream). In this case creating the DynamoDB table in KCL will fail. Instead the DynamoDB table should be created in your own AWS account.
#### Proposal

This PR adds a new serializable class `AWSCredentialPool` containing credentials for Kinesis, DynamoDB and CloudWatch that is passed as a parameter to `KinesisUtils.createStream()`. These credentials are then passed to the constructor of `KinesisClientLibConfiguration`.

@tdas @cfregly please review if this PR is OK.

See [SPARK-10969](https://issues.apache.org/jira/browse/SPARK-10969)

This contribution is @ajayrammohan and my original work and we license the work to the project under the project's open source license.

### CODE REVIEW

File: extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala [hunk 1]
```diff
@@ -221,13 +237,51 @@ private[kinesis] object KinesisTestUtils {
...
+      }
+      case Failure(e) =>
+        println(
+          s"""
+            |Could not load the AWS credentials for profile $profile.
+            |
+            |Credentialpool tests verify the ability to uses Kinesis stream, Dynamo DB and CloudWatch Metrics in
+            |separate AWS accounts. This requires a separate set of AWS credentials to be stored in system with the profile
+            |names "dynamoDB" and "cloudWatch" along with the default credentials. The test assumes that the Kinesis stream 
+            |in the  In absence of this only the underlying API will be tested.
```
Comment (soulcutter) [inline extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala]: This sentence looks like it got mangled during editing somehow.

**Comment by soulcutter** [comment]
I like the idea of being able to provide separate credentials for services, however it seems to me like a more-common use case is still going to be providing a single set of credentials for all services. It would be nice if the way to do so was still as straightforward as it is currently.

**Comment by kaklakariada** [comment]
@soulcutter Thank you very much for your review and sorry for the delay!
Please correct me if I am wrong, but in test [testKinesisStreamWithSimpleAwsCredentials](https://github.com/ajayrammohan/spark/blob/SPARK-10969/extras/kinesis-asl/src/test/java/org/apache/spark/streaming/kinesis/JavaKinesisStreamSuite.java#L50) we test, that you still can use a single set of credentials with the old API.

**Comment by soulcutter** [comment]
Welp I think you're right :)

**Comment by kaklakariada** [comment]
Solved with https://issues.apache.org/jira/browse/SPARK-19911 / #17250, see [this comment](https://issues.apache.org/jira/browse/SPARK-10969?focusedCommentId=16266374&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16266374).

**Pull request closed without merging**